Личный проект: сервер наносит первый удар

TwoKDev opened the pull request:
---
:mortar_board: [Личный проект: сервер наносит первый удар](https://up.htmlacademy.ru/nodejs/3/user/82178/tasks/12)

Commit:
module2-task3: replace relative -> absolute path

Commit:
module2-task3: install nodemon and @babel/node

Commit:
module2-task3: add command --server <port>
- add command --server <port>
- add npm script server
- change npm script start

Baristanko commented on src/service/cli/index.ts:
```diff
@@ -1,26 +1,32 @@
+const DEFAULT_COMMAND = commandHelp;
+
+export const processCLICommands = async (userArgs: string[]): Promise<void> => {
+  const [commandName, commandValue] = userArgs;
```
на данный момент реализация не поддерживает если у команды будет несколько аргументов, лучше сделать так:
```js
const [commandName, ...commandArgs] = userArgs;
...
command.run(...commandArgs);
```
Тогда все аргументы после названия команды передадутся в нее в отдельном параметре.

TwoKDev replied:
поправил

Baristanko commented on src/service/cli/server.ts:
```diff
@@ -0,0 +1,94 @@
+
+export const commandServer = {
+  name: `--server` as const,
+  async run(...args: string[]): Promise<void> {
```
если использовать способ что я описал выше с передачей параметров, то можно не использовать ...args здесь, а каждый параметр описать отдельно, будет намного понятнее сигнатура вызова функции

TwoKDev replied:
поправил

Baristanko commented on src/service/cli/server.ts:
```diff
@@ -0,0 +1,94 @@
+        sendResponse(response, HttpCode.NOT_FOUND, notFoundMessageText);
+      }
+
+      const publications: Publication[] = JSON.parse(mockContent);
```
например у нас в mock файле данные будут не в json-формате, здесь возникнет исключение, которое нигде не обрабатывается: нужно чтение файла с моками и преобразование полученных данных вынести в отдельную функцию и вызывать ее в блоке try

TwoKDev replied:
Ок, понял. Тоже эта часть смущала. Что запись и чтение/преобразование mock данных разбросано по разным модулям.

TwoKDev replied:
разбил логику генерации и работу с моками публикаций в отдельные модули.

Commit:
module2-task3 Разделил работу с данными публикации
- Вынес логику генерации в отдельный модуль publicationGenerate
- Вынес логику записи и чтения моковых данных в модуль publicationMock
